dags/analyzer_helper/tests/integration/test_discord_load_transformed_data.py

```diff
@@ -1,101 +1,146 @@
+from hivemind_etl_helpers.src.utils.mongo import MongoSingleton
 
 
 class TestDiscordLoadTransformedData(unittest.TestCase):
-    @classmethod
-    def setUpClass(cls):
-        cls.client = MongoSingleton.get_instance().client
-        cls.db = cls.client['discord_platform']
-        cls.collection = cls.db['rawmemberactivities']
-
-    @classmethod
-    def tearDownClass(cls):
-        cls.collection.delete_many({})
-        cls.client.close()
-
     def setUp(self):
+        self.client = MongoSingleton.get_instance().client
+        self.db = self.client['discord_platform']
+        self.collection = self.db['rawmemberactivities']
         self.collection.delete_many({})
         self.loader = DiscordLoadTransformedData('discord_platform')
 
     def tearDown(self):
         self.collection.delete_many({})
+        self.client.close()
 
     def test_load_data(self):
         processed_data = [
             {
-                'author_id': 'user123',
-                'date': datetime(2023, 1, 1),
-                'source_id': 'discord_platform',
-                'metadata': {
-                    'channel_id': 'channel456',
-                    'channel_name': 'general',
-                    'thread_id': 'thread123',
-                    'thread_name': 'discussion',
-                    'message_id': 'msg123',
-                    'is_generated_by_webhook': False
+                "author_id": "159985870458322944",
+                "date": datetime(2023, 3, 22, 18, 21, 0, 870000),
+                "source_id": "msg123",
+                "metadata": {
+                    "thread_id": "thread123",
+                    "channel_id": "1088165451651092635",
+                    "bot_activity": False,
                 },
-                'actions': [
+                "actions": [
                     {
-                        'name': 'message',
-                        'type': 'emitter'
+                        "name": "message",
+                        "type": "emitter"
                     }
                 ],
-                'interactions': [
+                "interactions": [
                     {
-                        'name': 'reply',
-                        'users_engaged_id': ['user789'],
-                        'type': 'receiver'
+                        "name": "reply",
+                        "users_engaged_id": ["159985870458322945"],
+                        "type": "emitter"
                     }
                 ]
+            },
+            {
+                "author_id": "159985870458322945",
+                "date": datetime(2023, 3, 22, 18, 21, 0, 870000),
+                "source_id": "msg123",
+                "metadata": {
+                    "thread_id": "thread123",
+                    "channel_id": "1088165451651092635",
+                    "bot_activity": False,
+                },
+                "actions": [
+                    {
+                        "name": "reply",
```

Comment:
reply is an interaction. On discord we would just have the `message` as an action.

Here's a definition for action and interactions

1. **Action:** Each activity that doesn't include a second person. i.e. messaging on discord with no reply, mention, or reaction received.
2. **Interaction**: Each activity that other people are participated in. i.e. mention, reaction, reply, ...